projects/union_path_integration/convergence_simulation.py

```diff
@@ -157,28 +181,96 @@ def doExperiment(cellDimensions, cellCoordinateOffsets, numObjects,
+  :return:
+  """
+  experiments = [{}]
+  for key, values in kwargs.items():
+    if type(values) is list:
+      new_experiments = []
+      for experiment in experiments:
+        for val in values:
+          new_experiment = copy(experiment)
+          new_experiment[key] = val
+          new_experiments.append(new_experiment)
+      experiments = new_experiments
+    else:
+      [a.__setitem__(key, values) for a in experiments]
+
+  numWorkers = cpu_count()
+  if numWorkers > 1:
+    pool = Pool(processes=numWorkers)
+    rs = pool.map_async(experimentWrapper, experiments, chunksize=1)
+    while not rs.ready():
+      remaining = rs._number_left
+      pctDone = 100.0 - (100.0*remaining) / len(experiments)
+      print "    =>", remaining, "experiments remaining, percent complete=",pctDone
+      time.sleep(5)
+    pool.close()  # No more work
+    pool.join()
+    result = rs.get()
+  else:
+    result = []
+    for arg in experiments:
+      result.append(doExperiment(arg))
+
+  # Pickle results for later use
+  results = [(arg,res) for arg, res in zip(experiments, result)]
+  with open(resultName,"wb") as f:
+    pkl.dump(results,f)
+
+  return results
 
 
 if __name__ == "__main__":
   parser = argparse.ArgumentParser()
-  parser.add_argument("--numObjects", type=int, required=True)
+  parser.add_argument("--numObjects", type=int, nargs="+", required=True)
   parser.add_argument("--numUniqueFeatures", type=int, required=True)
   parser.add_argument("--locationModuleWidth", type=int, required=True)
-
   parser.add_argument("--coordinateOffsetWidth", type=int, default=7)
+  parser.add_argument("--useNoise", action="store_true")
   parser.add_argument("--useTrace", action="store_true")
+  parser.add_argument("--resultName", type = str, default = "results.pkl")
+  parser.add_argument("--method", type = str, default = "narrowing")
 
   args = parser.parse_args()
 
   numOffsets = args.coordinateOffsetWidth
-  cellCoordinateOffsets = [i * (0.998 / (numOffsets-1)) + 0.001
-                           for i in xrange(numOffsets)]
-  doExperiment(
+  cellCoordinateOffsets = tuple([i * (0.998 / (numOffsets-1)) + 0.001 for i in xrange(numOffsets)])
+
+  if args.useNoise:
+    noiseFactor = list(np.arange(0, 5, .5))
+    moduleNoiseFactor = list(np.arange(0, 5, .5))
+  else:
+    noiseFactor = moduleNoiseFactor = 0
+
+  if "both" in args.method:
+    args.method = ["narrowing", "corners"]
+
+
+  run_multiprocess_noise_experiment(args.resultName,
     cellDimensions=(args.locationModuleWidth, args.locationModuleWidth),
     cellCoordinateOffsets=cellCoordinateOffsets,
     numObjects=args.numObjects,
     featuresPerObject=10,
     objectWidth=4,
     numFeatures=args.numUniqueFeatures,
     useTrace=args.useTrace,
+    use_noise=args.useNoise,
+    noise_factor=noiseFactor,
+    module_noise_factor=moduleNoiseFactor,
+    method=args.method,
   )
+
+
+# Widths = [5, 10, 20]
```

Comment:
If you want to include some notes like this you could put them in a README.md in the project directory